### ISSUE TITLE
allow setting endpoints for autoscaling

### ISSUE THREAD

**Issue by bardielle** [terraform-providers/terraform-provider-aws#4967]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v1.23.0

### Affected Resource(s)

* website/docs/index.html.markdown
* provider.go
* config.go

### Terraform Configuration Files

aws-provider.tf file
```
provider "aws" {
  access_key = "access_key"
  secret_key = "secret_key"

endpoints {
    ec2 = "https://10.0.0.1/api/v2/aws/ec2"
    autoscaling = "https://10.0.0.1/api/v2/aws/asg"
  }

  insecure = "true"
  skip_metadata_api_check = true
  skip_credentials_validation = true
  region = "eu-west-1"
}

```
N/A

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Debug Output

sudo ./terraform apply
Error: provider.aws: endpoints.0: invalid or unknown key: autoscaling

### Panic Output

N/A

### Expected Behavior

Terraform provider should allow users to set autoscaling endpoint

### Actual Behavior

Terraform provider doesn't allow users to set autoscaling endpoint

### Steps to Reproduce

1.  Set aws-provider autoscaling endpoint
2. Run `terraform apply`

### Important Factoids

N/A

### References

This is a similar bug to
https://github.com/terraform-providers/terraform-provider-aws/pull/2795/files

**Comment by bflad** [comment]
This has been merged into master via #4970 and will release with version 1.25.0 of the AWS provider, likely middle of this week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.25.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1250-june-27-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
